components/menu/menu.js

```diff
@@ -0,0 +1,354 @@
+import '../icons/icon.js';
+import './menu-item-return.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { HierarchicalViewMixin } from '../hierarchical-view/hierarchical-view-mixin.js';
+
+class Menu extends HierarchicalViewMixin(LitElement) {
+
+	static get properties() {
+		return {
+			active: { type: Boolean, reflect: true },
+			label: { type: String },
+			role: { type: String, reflect: true }
```

Comment:
Is this needed?  Does it conflict with the native property?

Reply:
We could remove this and set the `tabindex` (in menu-item) and `role` in the constructor using `setAttribute`

Reply:
I did some more looking into this, and because of how we are timing things, `role` and `tabindex` need to be in properties (I haven't seen issue with conflicting with the native property and using `setAttribute` to update `tabindex` for a menu item works fine, and it seems to work either way it's set) - there might be a better way to do this though. 

For example, in `menu.js` we look for role being menu item on items to see if they are visible, and if in `menu-item-mixin` we do `setAttribute` in `firstUpdated`, it's too late for `menu` to see it and so the code that runs doesn't pick up the menu items (we could fire an event when the menu-item is updated, but then we end up firing an event for each menu-item and that seems excessive). We can't do `setAttribute` in `constructor`.

Reply:
^ This above explanation is more applicable to `menu-item-mixin.js`. If we use `this.setAttribute('role', 'menu')` for this case in `menu.js` and remove this role property, it works fine.